staging/src/k8s.io/csi-translation-lib/plugins/openstack_cinder.go

```diff
@@ -144,6 +144,12 @@ func (t *osCinderCSITranslator) TranslateCSIPVToInTree(pv *v1.PersistentVolume)
 		ReadOnly: csiSource.ReadOnly,
 	}
 
+	// translate CSI topology to In-tree topology for rollback compatibility.
+	// It is not possible to guess Cinder Region from the Zone, therefore leave it empty.
+	if err := translateTopologyFromCSIToInTree(pv, CinderTopologyKey, nil); err != nil {
```

Comment:
Can you add a unit test in the intree_volume_test `TestTranslateTopologyFromCSIToInTree` to make sure the translation is as expected?
I am less familiar with Cinder, does cinder not have a region topology label at all?

Reply:
Oh sorry, I saw the PR description now. I think if there is absolutely no way we can retrieve the info from the CSI PV object, then I think it is better that we double check with OpenStack team to see if a missing region topology label some time later is okay. Because when the label admission controller is removed, this will be a regression from user perspective. @msau42 for any other thoughts.

Reply:
Yes it has region label.

Reply:
For the other clouds, region is also just for labeling, so not having it will still function from a Kubernetes perspective. The only issue is if users were depending on that region label for other purposes.

This could potentially be addressed by deploying an admission webhook with the CSI driver or having the CSI driver return region as part of the topology.

Reply:
Added unit test as suggested, however, it's basically the same as `regionParser is missing and only zone labels get generated`, only with Cinder labels.